plugins/parcelize/parcelize-compiler/src/org/jetbrains/kotlin/parcelize/ParcelizeResolveExtension.kt

```diff
@@ -197,16 +201,11 @@ val RAW_VALUE_ANNOTATION_FQ_NAMES = listOf(
 internal val PARCELER_FQNAME = FqName("kotlinx.parcelize.Parceler")
 internal val OLD_PARCELER_FQNAME = FqName("kotlinx.android.parcel.Parceler")
 
-val ClassDescriptor.isParcelize: Boolean
-    get() {
-        for (fqName in PARCELIZE_CLASS_FQ_NAMES) {
-            if (this.annotations.hasAnnotation(fqName)) {
-                return true
-            }
-        }
+val ClassDescriptor.hasParcelizeAnnotation: Boolean
+    get() = PARCELIZE_CLASS_FQ_NAMES.any(annotations::hasAnnotation)
 
-        return false
-    }
+val ClassDescriptor.isParcelize: Boolean
+    get() = hasParcelizeAnnotation || getSuperClassNotAny()?.takeIf(DescriptorUtils::isSealedClass)?.hasParcelizeAnnotation == true
```

Comment:
Probably also needs to handle sealed interfaces

Reply:
Done